erica/api/utils.py

```diff
@@ -0,0 +1,51 @@
+from uuid import UUID
+
+from opyoid import Injector
+
+from erica.api.ApiModule import ApiModule
+from erica.api.v2.responses.model import JobState
+from erica.application.EricaAuftrag.EricaAuftragService import EricaAuftragServiceInterface
+from erica.domain.Shared.Status import Status
+
+
+def map_status(status: Status):
+    """
+    Mapper from internal rq job state to API job state.
+            Parameters:
+                    status (Status): the internal rq job state.
+            Returns:
+                    (JobState): the corresponding API job state.
+    """
+    switcher = {
+        Status.new: JobState.PROCESSING,
+        Status.scheduled: JobState.PROCESSING,
+        Status.processing: JobState.PROCESSING,
+        Status.failed: JobState.FAILURE,
+        Status.success: JobState.SUCCESS
+    }
+    return switcher.get(status)
+
+
+def generate_error_response(errorcode=-1, errormessage="API resource not yet implemented."):
+    """
+    Generator of error response for the API v2.
+            Parameters:
+                    errorcode (int): the error code number.
+                    errormessage (str): the error message.
+            Returns:
+                    (dict): dict with error respones object.
+    """
+    error_response = {"errorCode": errorcode,
+                      "errorMessage": errormessage
+                      }
+    return error_response
+
+
+injector = Injector([
+    ApiModule(),
+])
+
+
+def get_erica_request(job_id: UUID):
+    freischalt_code_service: EricaAuftragServiceInterface = injector.inject(EricaAuftragServiceInterface)
+    return freischalt_code_service.get_status(job_id)
```

Comment:
You mean testing that functionality in a separate way? Because we are already testing if an entity is found or not within the integration tests for the queue.

And regarding the use of a mapper you mean that the function maps the state of the entity to the state of the API? But we need the whole entity in the API methods since we are not only mapping the state but also extracting the data from the result column

Reply:
> You mean testing that functionality in a separate way? Because we are already testing if an entity is found or not within the integration tests for the queue.

Yes, testing that function on a unit test basis not on an integration level. However, I thought about it a bit more and you are right. This function makes only sense to be tested on an integration level and then it is probably better to test the api-level functions.

> And regarding the use of a mapper you mean that the function maps the state of the entity to the state of the API? But we need the whole entity in the API methods since we are not only mapping the state but also extracting the data from the result column

Totally right. Maybe we should rename the function in get_status to something else or perform the mapping already there.

Reply:
Yes, we talked about it. It's fine.